admyral/server/endpoints/secret_endpoints.py

```
@@ -8,22 +8,35 @@
-            for key, val in secret.secret.items()
-        ]
-    )
-    await get_secrets_manager().set(authenticated_user.user_id, secret)
+) -> SecretMetadata:
+    secret.secret = handle_secret_values(secret.secret)
+    return await get_secrets_manager().set(authenticated_user.user_id, secret)
+
+
+@router.post("/update", status_code=status.HTTP_201_CREATED)
+async def update_secret(
+    secret: Secret, authenticated_user: AuthenticatedUser = Depends(authenticate)
```

Ruff

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)